documentation editorial pass (#129)
astrolicious/astro-theme-provider: docs/src/content/docs/core-concepts.mdx

---
title: Core Concepts
description: Core concepts of Astro Theme Provider
---

import { FileTree } from '@astrojs/starlight/components';

This page aims to explain how Astro Theme Provider works at a fundamental level for _theme authors_.
If you are interested in learning about what Astro Theme Provider is and why you want to use it, you can read more here: [Why?](/why)
If you need specific advice about how to author a theme, check out the guides under "Conventions and Techniques" in the left sidebar.
If you are a _theme user_ please refer to [User API](/reference/user) and your theme's documentation.

## Authoring a Theme

Using Astro Theme Provider to author a theme integration is simple, but there are a few core concepts that one should know before getting started.

### Project Structure

Themes are developed inside a [pnpm](https://pnpm.io/) monorepository.

> <FileTree>
> - /
>   - package/
> 	- playground/
> </FileTree>

Reviewer: ```suggestion
<FileTree>
- /
  - package/
  	- ...
  - playground/
  	- ...
  - package.json
  - pnpm-lock.yaml
  - pnpm-workspace.yaml
  - tsconfig.json
</FileTree>
```
This is an example of a simple pnpm monorepo for a theme

		- pages/
		- styles/
	- index.ts
	- package.json
	- README.md
</FileTree>

Theme integrations must be authored alongside a playground.
The playground is responsible for generating types during development and enables previewing changes made to the theme.

<FileTree>
- playground/
	- public/
	- src/
	- astro.config.mjs
	- package.json
	- tsconfig.json
</FileTree>

### Theme Configuration

> Theme integrations are published with a built-in config file that can be modified by a theme user.

Reviewer: ```suggestion
Theme integrations can be modified by a theme user from inside the Astro config.
```
Theme are configured from inside the Astro config

Author: I feel like "the Astro config" is a confusing way to phrase this. Would it be correct to say "the Astro config object" or something?

Reviewer: Ya that works, I am not married to my suggestion so whatever you think is best

Author: Okay, I changed this section a bit to clarify things. I think the "Astro config object" language is fine if we want to draw a distinction between the file and particular part of the code.
